lib/proposed/character_validator.rb

```diff
@@ -0,0 +1,17 @@
+class CharacterValidator
+
+  def initialize(character)
```

Comment:
Walidacja jest wyrzucona do klasy Validatora, łatwiej to przetestować i utworzyć różne warianty do characteru, dzięki czemu, możemy sobie testować walidację w izolacji. Poza tym jest tu oddzielenie warstwy modelu od jego walidacji.